libp2p/autotls/autotls.nim

```diff
@@ -0,0 +1,245 @@
+# Nim-Libp2p
+# Copyright (c) 2025 Status Research & Development GmbH
+# Licensed under either of
+#  * Apache License, version 2.0, ([LICENSE-APACHE](LICENSE-APACHE))
+#  * MIT license ([LICENSE-MIT](LICENSE-MIT))
+# at your option.
+# This file may not be copied, modified, or distributed except according to
+# those terms.
+
+{.push raises: [].}
+{.push public.}
+
+import
+  net,
+  results,
+  chronos,
+  chronicles,
+  bearssl/rand,
+  bio,
+  json,
+  chronos/apps/http/httpclient
+from std/times import DateTime
+
+import ./acme
+import ./peeridauth
+import ./utils
+import ../nameresolving/dnsresolver
+import ../wire
+import ../crypto/crypto
+import ../peerinfo
+import ../utils/heartbeat
+
+logScope:
+  topics = "libp2p autotls"
+
+const
+  DefaultDnsServers =
+    @[
+      initTAddress("1.1.1.1:53"),
+      initTAddress("1.0.0.1:53"),
+      initTAddress("[2606:4700:4700::1111]:53"),
```

Comment:
Interestingly enough, `dnsclient` uses 8.8.8.8. I kinda prefer cloudflare like you added here, but wonder if we should try to unify both?